sigle/src/modules/userCard/UserCard.tsx

```diff
@@ -27,37 +43,71 @@ const ProfileImageContainer = styled('a', {
   justifyContent: 'center',
   br: '$1',
   overflow: 'hidden',
-  width: 38,
-  height: 38,
+  width: 50,
+  height: 50,
   flex: 'none',
+
+  '@md': {
+    width: 38,
+    height: 38,
+  },
 });
 
 const ProfileImage = styled('img', {
+  maxWidth: 50,
+  maxHeight: 50,
+
   width: 'auto',
   height: '100%',
-  maxWidth: 38,
-  maxHeight: 38,
   objectFit: 'cover',
+
+  '@md': {
+    maxWidth: 38,
+    maxHeight: 38,
+  },
+});
+
+const UserCardTitle = styled(Typography, {
+  maxWidth: 160,
+  display: 'block',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+
+  '@md': {
+    maxWidth: 600,
+  },
 });
 
 const UserCardDescription = styled(Typography, {
+  maxWidth: 200,
   mt: '$1',
+  pr: '$2',
+  display: 'block',
   overflow: 'hidden',
-  display: '-webkit-box',
-  '-webkit-line-clamp': 2,
-  '-webkit-box-orient': 'vertical',
-  typographyOverflow: 'ellipsis',
-  maxWidth: 600,
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+
+  '@md': {
+    maxWidth: 600,
+    whiteSpace: 'normal',
+    display: '-webkit-box',
+    '-webkit-line-clamp': 2,
+    '-webkit-box-orient': 'vertical',
+    typographyOverflow: 'ellipsis',
+  },
 });
 
 interface UserCardProps {
   address: string;
 }
 
 export const UserCard = ({ address }: UserCardProps) => {
+  const { resolvedTheme } = useTheme();
   const { user, isLegacy } = useAuth();
   const { mutate: followUser } = useUserFollow();
   const { mutate: unfollowUser } = useUserUnfollow();
+  const { data: userInfoByAddress } = useGetUserByAddress(address);
```

Comment:
@gregogun can we separate the badge logic to another pr? This means we will be sending 50+ requests to the server (one for each user) and will hit rate limit fast. Will need to make some server changes to get it to work fine on prod. So let's move it to another pr to merge a bit later :)